sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -233,9 +243,17 @@ def send(self, event_data, partition_key=None, timeout=None):
                 if partition_key:
                     event_data = _set_partition_key(event_data, partition_key)
                 wrapper_event_data = EventDataBatch._from_batch(event_data, partition_key)  # pylint: disable=protected-access
+            for internal_message in wrapper_event_data.message._body_gen:  # pylint: disable=protected-access
+                self._client._trace_message(child, internal_message)  # pylint: disable=protected-access
         wrapper_event_data.message.on_send_complete = self._on_outcome
         self._unsent_events = [wrapper_event_data.message]
-        self._send_event_data_with_retry(timeout=timeout)
+
+        if span_impl_type is not None:
+            with child:
+                self._client._add_span_request_attributes(child)  # pylint: disable=protected-access
+                self._send_event_data_with_retry(timeout=timeout)
+        else:
+            self._send_event_data_with_retry(timeout=timeout)
```

Comment:
both if and else has this statement. How about removing the else block and pull the same statement in "if" out of "if"?